### PR TITLE
doc: extend beginning development with step to build in terminal before IDE

### PR DESCRIPTION
**Pull request by romani**
just based on multiple questions in  gitter on how to build our project.

### CODE REVIEW

**Comment by strkkk** [comment]
Github, generate site

File: src/xdocs/beginning_development.xml [hunk 1]
```diff
@@ -31,6 +31,14 @@
             git clone git@github.com:you_user_name/checkstyle.git
         </source>
       </div>
+      <p>
+        4. Before opening project in IDEA do build in terminal by running in repository root folder
+      </p>
+      <div class="wrap-content">
+        <source>
+            mvn install
```
Comment (strkkk) [inline src/xdocs/beginning_development.xml:39]: this will require contributor to have maven installed (see item 1)

**Review by strkkk — CHANGES_REQUESTED** [review]

**Comment by strkkk** [comment]
also, shouldnt this page be edited as well to resolve all questions? 
https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/985e42d_2021070331/idea.html

because it is a bit weird to see mention of idea on `beginning development` page.

File: src/xdocs/beginning_development.xml (hunk 1 above)
Comment (romani) [reply src/xdocs/beginning_development.xml:39]: mvn is added.
image

**Comment by romani** [comment]
Github, generate site

**Comment by romani** [comment]
> shouldnt this page be edited as well to resolve all questions?

I am not sure, but we can try. If you have list of question, please create issue, if you know how to improve shoot a PR.

**Review by strkkk — APPROVED** [review]

File: src/xdocs/beginning_development.xml (hunk 1 above)
Comment (rnveach) [inline src/xdocs/beginning_development.xml:39]: I am more curious why `mvn install` is needed. I never installed checkstyle snapshot outside of regression and never had any issues. Should this be something more like `mvn verify test`?

Comment (romani) [reply src/xdocs/beginning_development.xml:39]: It runs all phases of maven.
If this command pass, everything that contributor might run is ok and doable at command line.

Comment (romani) [reply src/xdocs/beginning_development.xml:39]: >mvn verify test

This is not correct command, it runs tests twice.

I just tried to be very simple in my command.
`install` is required for many no-error CI executions

Comment (rnveach) [reply src/xdocs/beginning_development.xml:39]: Ok.

Comment (pbludov) [reply src/xdocs/beginning_development.xml:39]: maven 3.0.1 works with any JDK version starting from 7. But there is some issues.
The recommended version of maven is `3.8.x` for JDK17 and  `3.6.x` for JDK 11

**Review by pbludov — APPROVED** [review]

File: src/xdocs/beginning_development.xml (hunk 1 above)
Comment (romani) [reply src/xdocs/beginning_development.xml:39]: I used version from our pom

File: src/xdocs/beginning_development.xml [hunk 2]
```diff
@@ -31,6 +31,14 @@
             git clone git@github.com:you_user_name/checkstyle.git
         </source>
       </div>
+      <p>
+        4. Before opening project in IDEA do build in terminal by running in repository root folder
```
Comment (rnveach) [inline src/xdocs/beginning_development.xml:35]: `IDEA` => `IDE`.

This is not IDEA specific instructions, so we should make it clear to the user it is for any IDE they are using and not to skip if they are not using IDEA.

**Review by rnveach — CHANGES_REQUESTED** [review]

File: src/xdocs/beginning_development.xml (hunk 2 above)
Comment (romani) [reply src/xdocs/beginning_development.xml]: fixed.

**Comment by romani** [comment]
Github, generate website

**Review by rnveach — APPROVED** [review]

**Pull request merged**